### PR TITLE
feat(headless): stageName parameter added to loadCaseNextStage case assist event

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4423](https://coveord.atlassian.net/browse/SFINT-4423)

- New metadata field `stage` added to the action `logCaseNextStage` that dispatches a `ticket_next_stage` event, this optional field will contain the information about the name of the stage from where the ticket_next_stage event was fired.

SFINT-4423

### CODE REVIEW

File: packages/headless/src/features/case-assist/case-assist-analytics-actions-loader.ts [hunk 1]
```diff
@@ -33,9 +33,12 @@ export interface CaseAssistAnalyticsActionCreators {
   /**
    * Creates a Case Assist event for when the user moves to the next stage.
    *
+   * @param stageName - The current stage name.
    * @returns A dispatchable action.
    */
-  logCaseNextStage(): AsyncThunkAction<
+  logCaseNextStage(
+    stageName?: string
```
Comment (ThibodeauJF) [inline packages/headless/src/features/case-assist/case-assist-analytics-actions-loader.ts:40]: Is it possible that more options will be added to this in the future? If yes, having an object of options instead will allow to add parameters more easily.

File: packages/headless/src/features/case-assist/case-assist-analytics-actions.ts [hunk 2]
```diff
@@ -13,13 +13,15 @@ export const logCaseStart = makeCaseAssistAnalyticsAction(
...
-      ticket: caseAssistCaseSelector(state),
-    })
-);
+export const logCaseNextStage = (stageName?: string) =>
+  makeCaseAssistAnalyticsAction(
+    'analytics/caseAssist/case/nextStage',
+    (client, state) =>
+      client.logMoveToNextCaseStep({
+        ticket: caseAssistCaseSelector(state),
+        stage: stageName,
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/headless/src/features/case-assist/case-assist-analytics-actions.ts]: I already specified in coveo.analytics.js that the field will be named stage 😬

File: packages/headless/package-lock.json [hunk 3]
```diff
@@ -3,5512 +3,6 @@
   "version": "1.55.0",
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/headless/package-lock.json:3]: Thank you! I'll try this👌

[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/headless/package-lock.json]: It worked perfectly, Thank you!

**Pull request merged**